### PR TITLE
set "noImplicitAny" to true and fix the problems

### PR DESCRIPTION
**Pull request by oltolm**
I did some work on code-debug. This is a first PR in a series. I set `noImplicitAny` to true and fixed the problems. I also found a bug in `gdb_expansion.ts` line 226. Not sure if I fixed it correctly.

### CODE REVIEW

File: src/backend/gdb_expansion.ts [hunk 1]
```diff
@@ -223,7 +224,7 @@ export function expandValue(variableCreate: Function, value: string, root: strin
 				val = "Object@" + val;
 			}
 		} else if (typeof val == "string" && val.startsWith("@0x")) {
-			ref = variableCreate(getNamespace("*&" + name.substr));
+			ref = variableCreate(getNamespace("*&" + name.substring(1)));
```
Comment (WebFreak001) [inline src/backend/gdb_expansion.ts:227]: ah nice catch, do you know why you remove the first character here? It looks like maybe just passing `name` instead might be correct instead. Sadly there is no test for this to actually verify this.

**Review by WebFreak001 — APPROVED** [review]
awesome! Thank you very much for these much needed improvements

**Comment by GitMensch** [comment]
I'm eager to see the rest of the patch series 💪 .

File: src/backend/gdb_expansion.ts (hunk 1 above)
Comment (oltolm) [reply src/backend/gdb_expansion.ts:227]: That was just a guess. Feel free to change it.

**Pull request merged**